graphql_compiler/cost_estimation/interval.py

```diff
@@ -0,0 +1,91 @@
+import datetime
+from typing import Generic, Optional, TypeVar
+
+
+IntervalDomain = TypeVar("IntervalDomain", int, str, datetime.date, datetime.datetime)
+
+
+class Interval(Generic[IntervalDomain]):
+    def __init__(
+        self, lower_bound: Optional[IntervalDomain], upper_bound: Optional[IntervalDomain]
+    ) -> None:
+        """Initialize an interval. A value of None means unbounded."""
+        self.lower_bound: Optional[IntervalDomain] = lower_bound
+        self.upper_bound: Optional[IntervalDomain] = upper_bound
+
+    def is_empty(self) -> bool:
+        """Return whether the interval is empty"""
+        if self.lower_bound is None or self.upper_bound is None:
+            return False
+        return self.lower_bound > self.upper_bound
```

Comment:
Presumably this means that `lower_bound == upper_bound` is an interval with a singular value in it? I.e. both bounds are inclusive? Would be nice to call that out specifically.

Reply:
Called it out in the init spec.